force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -122,22 +131,25 @@ public with sharing class ClassesObjectsMethods {
      * @return A string containing the names of scientists that contain the letter 'g'.
      */
     public static String findTheScientist() {
-        // The list of scientists' names
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
-
         // The variable to store the concatenated string
-
+        String namesAnswer = '';
         // Loop through the list of scientists' names
-
+        for (String name : scientistNames) {
             // If the name doesn't contain the letter 'G', skip this iteration
-
+            if (name.toLowerCase().contains('g')) {
             // Add the name to the result string, followed by a comma
-
-        // Remove the last comma from the result string
-
-        return null; // Replace null with the variable you used to store the result
+                namesAnswer+= name + ',';
```

Comment:
Add them to a List first, then use String.join to make the CSV when you are done adding names. This is the most performant and typical approach. You won't have to remove a trailing comma either.